src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs

```diff
@@ -256,38 +259,37 @@ internal TabItem AddTabItem(IViewExtension viewExtension, ContentControl content
         private void CloseTab(object sender, RoutedEventArgs e)
         {
             string tabName = (sender as Button).CommandParameter.ToString();
-            CloseTab(tabName);
+
+            TabItem tabitem = TabItems.OfType<TabItem>().SingleOrDefault(n => n.Name == tabName);
+            CloseTab(tabitem);
         }
 
         /// <summary>
         /// Close tab by its name
         /// </summary>
-        /// <param name="tabName">tab name</param>
-        private void CloseTab(string tabName)
+        /// <param name="tabitem">tab item</param>
+        private void CloseTab(TabItem tabitem)
         {
-            TabItem tab = tabDynamic.SelectedItem as TabItem;
+            TabItem tabToBeRemoved = tabitem;
 
-            if (tab != null)
-            {
-                // get the selected tab
-                TabItem selectedTab = tabDynamic.SelectedItem as TabItem;
+            // get the selected tab
+            TabItem selectedTab = tabDynamic.SelectedItem as TabItem;
 
-                // clear tab control binding and bind to the new tab-list. 
-                tabDynamic.DataContext = null;
-                TabItems.Remove(tab);
-                TabItems = TabItems;
-                tabDynamic.DataContext = TabItems;
+            // clear tab control binding and bind to the new tab-list. 
+            tabDynamic.DataContext = null;
+            TabItems.Remove(tabToBeRemoved);
+            TabItems = TabItems;
```

Comment:
??

Reply:
Removed this unnecessary line. Thanks @aparajit-pratap